packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -43,20 +45,53 @@ import {
 } from './gridVirtualizationSelectors';
 import { EMPTY_RENDER_CONTEXT } from './useGridVirtualization';
 
-export const EMPTY_DETAIL_PANELS = Object.freeze(new Map<GridRowId, React.ReactNode>());
-
-export type VirtualScroller = ReturnType<typeof useGridVirtualScroller>;
+const MINIMUM_COLUMN_WIDTH = 50;
 
 interface PrivateApiWithInfiniteLoader
   extends GridPrivateApiCommunity,
     GridInfiniteLoaderPrivateApi {}
 
+export type VirtualScroller = ReturnType<typeof useGridVirtualScroller>;
+
+type ScrollPosition = { top: number; left: number };
+enum ScrollDirection {
+  NONE,
+  UP,
+  DOWN,
+  LEFT,
+  RIGHT,
+}
+
+const EMPTY_SCROLL_POSITION = { top: 0, left: 0 };
+
+export const EMPTY_DETAIL_PANELS = Object.freeze(new Map<GridRowId, React.ReactNode>());
+
+const createScrollCache = (
+  rowBufferPx: number,
+  columnBufferPx: number,
+  verticalBuffer: number,
+  horizontalBuffer: number,
+) => ({
+  direction: ScrollDirection.NONE,
+  buffer: bufferForDirection(
+    ScrollDirection.NONE,
+    rowBufferPx,
+    columnBufferPx,
+    verticalBuffer,
+    horizontalBuffer,
+  ),
+});
+type ScrollCache = ReturnType<typeof createScrollCache>;
+
+const isJSDOM = typeof window !== 'undefined' ? /jsdom/.test(window.navigator.userAgent) : false;
+
 export const useGridVirtualScroller = () => {
   const apiRef = useGridPrivateApiContext() as React.MutableRefObject<PrivateApiWithInfiniteLoader>;
   const rootProps = useGridRootProps();
   const visibleColumns = useGridSelector(apiRef, gridVisibleColumnDefinitionsSelector);
-  const enabled = useGridSelector(apiRef, gridVirtualizationEnabledSelector);
-  const enabledForColumns = useGridSelector(apiRef, gridVirtualizationColumnEnabledSelector);
+  const enabled = useGridSelector(apiRef, gridVirtualizationEnabledSelector) && !isJSDOM;
+  const enabledForColumns =
+    useGridSelector(apiRef, gridVirtualizationColumnEnabledSelector) && !isJSDOM;
```

Comment:
This PR broke about 30 tests in jsdom. It was this or adding `disableVirtualization={isJSDOM}` in those 30 tests.

Reply:
> This PR broke about 30 tests in jsdom.

Do you have an idea why?

Reply:
No, but seeing how this single change fixed them all it's virtualization for sure.